js/src/annotations/osd-svg-overlay.js

```diff
@@ -0,0 +1,544 @@
+(function($) {
+  $.getTools = function() {
+    if (this.svgOverlayTools) {
+      return this.svgOverlayTools;
+    }
+    this.svgOverlayTools = [new $.Rectangle(), new $.Freehand(), new $.Polygon(), new $.Ellipse(), new $.Pin()];
+    return this.svgOverlayTools;
+  };
+
+  OpenSeadragon.Viewer.prototype.svgOverlay = function(windowObj) {
+    return new $.Overlay(this, windowObj);
+  };
+
+  $.Overlay = function(viewer, windowObj) {
+    jQuery.extend(this, {
+      disabled: false,
+      window: windowObj,
+      windowId: windowObj.windowId,
+      commentPanel: null,
+      mode: '', // Possible modes: 'create', 'translate', 'deform', 'edit' and '' as default.
+      draftPaths: [],
+      editedPaths: [],
+      hoveredPath: null,
+      path: null,
+      segment: null,
+      latestMouseDownTime: -1,
+      doubleClickReactionTime: 300,
+      hitOptions: {
+        fill: true,
+        stroke: true,
+        segments: true,
+        tolerance: 5
+      }
+    });
+
+    // Initialization of overlay object.
+    this.tools = $.getTools();
+    this.currentTool = null;
+    // Default colors.
+    this.strokeColor = 'red';
+    this.fillColor = 'green';
```

Comment:
These default values should be set in settings.js so a developer can change them through the configuration.  By default, fillColor should be empty, and strokeColor should be ```deepSkyBlue```, to use the values we used with the previous annotation creation functionality.